chrome/content/zotero/integration/quickFormat.js

```diff
@@ -1397,120 +1768,136 @@ var Zotero_QuickFormat = new function () {
-
-					var nodeRange = qfiDocument.createRange();
-					nodeRange.selectNode(bubble);
-					nodeRange.collapse(!right);
-
-					var selection = qfiWindow.getSelection();
-					selection.removeAllRanges();
-					selection.addRange(nodeRange);
+					return;
 				}
-			})
-		} else if(keyCode === event.DOM_VK_UP && referencePanel.state === "open") {
-			locatorLocked = true;
-			var selectedItem = referenceBox.selectedItem;
-
-			var previousSibling;
-			
-			// Seek the closet previous sibling that is not disabled
-			while((previousSibling = selectedItem.previousSibling) && previousSibling.hasAttribute("disabled")) {
-				selectedItem = previousSibling;
 			}
-			// If found, change to that
-			if(previousSibling) {
-				referenceBox.selectedItem = previousSibling;
-				
-				// If there are separators before this item, ensure that they are visible
-				var visibleItem = previousSibling;
-
-				while(visibleItem.previousSibling && visibleItem.previousSibling.hasAttribute("disabled")) {
-					visibleItem = visibleItem.previousSibling;
-				}
-				referenceBox.ensureElementIsVisible(visibleItem);
-			};
+			// Tab places focus on the existing input or creates a new one in the end
+			refocusLastInput();
+			
 			event.preventDefault();
-		} else if(keyCode === event.DOM_VK_DOWN) {
-			locatorLocked = true;
-			if((Zotero.isMac ? event.metaKey : event.ctrlKey)) {
-				// If meta key is held down, show the citation properties panel
-				var bubble = _getSelectedBubble();
-
-				if(bubble) _showCitationProperties(bubble);
-				event.preventDefault();
-			} else if (referencePanel.state === "open") {
-				var selectedItem = referenceBox.selectedItem;
-				var nextSibling;
-				
-				// Seek the closet next sibling that is not disabled
-				while((nextSibling = selectedItem.nextSibling) && nextSibling.hasAttribute("disabled")) {
-					selectedItem = nextSibling;
-				}
-				
-				// If found, change to that
-				if(nextSibling){
-					referenceBox.selectedItem = nextSibling;
-					referenceBox.ensureElementIsVisible(nextSibling);
-				};
-				event.preventDefault();
-			}
-		} else {
+		}
+		else {
 			isPaste = false;
-			_resetSearchTimer();
 		}
 	});
 	
 	/**
 	 * Adds a dummy element to make dragging work
 	 */
 	function _onBubbleDrag(event) {
-		dragging = event.currentTarget;
-		event.dataTransfer.setData("text/plain", '<span id="zotero-drag"/>');
+		bubbleDragged = true;
+		this.style.cursor = "grabbing";
+		// Sometimes due to odd mozilla drag-drop behavior, the dragend event may not fire
+		// so the element will get stuck with the id. Clean it up on next dragstart if it happens.
+		let lastDragged = document.querySelector("#dragged-bubble");
+		if (lastDragged) {
+			lastDragged.removeAttribute("id");
+		}
+		setTimeout(() => {
+			this.setAttribute("id", "dragged-bubble");
+		});
+		let index = _getBubbleIndex(this);
+		event.dataTransfer.setData("zotero/citation_bubble", index);
+		this.setAttribute("original-index", index);
```

Comment:
Should use `dataset`. In the past I've had issues with setting non-standard attributes on elements, so better not go there.

Reply:
Makes sense, I'll keep that in mind for the future. In this case, I got rid of it completely - there was a boolean variable used to keep track if dragging is in progress (for window resizing). Might as well make it more useful and store an original index there